pkg/deploy/api/validation/validation.go

```diff
@@ -186,6 +188,17 @@ func validateRollingParams(params *deployapi.RollingDeploymentStrategyParams) fi
 			errs = append(errs, fielderrors.NewFieldInvalid("updatePercent", *params.UpdatePercent, "must be between 1 and 100 or between -1 and -100 (inclusive)"))
 		}
 	}
+	// Most of this is lifted from the upstream experimental deployments API. We
+	// can't reuse it directly yet, but no use reinventing the logic, so copy-
+	// pasted and adapted here.
+	errs = append(errs, ValidatePositiveIntOrPercent(params.MaxUnavailable, "maxUnavailable")...)
+	errs = append(errs, ValidatePositiveIntOrPercent(params.MaxSurge, "maxSurge")...)
+	if getIntOrPercentValue(params.MaxUnavailable) == 0 && getIntOrPercentValue(params.MaxSurge) == 0 {
+		// Both MaxSurge and MaxUnavailable cannot be zero.
+		errs = append(errs, fielderrors.NewFieldInvalid("maxUnavailable", params.MaxUnavailable, "cannot be 0 when maxSurge is 0 as well"))
+	}
+	// Validate that MaxUnavailable is not more than 100%.
+	errs = append(errs, IsNotMoreThan100Percent(params.MaxUnavailable, "maxUnavailable")...)
```

Comment:
Shouldn't we also validate the use of the deprecated `UpdatePercent` against the use of `MaxUnavailable` / `MaxSurge` ?
```go
if params.UpdatePercent != nil && (params.MaxUnavailable != nil || params.MaxSurge != nil) {
```